### PR TITLE
Ehn integration windows

### PR DESCRIPTION
**Pull request by pburdet**
Add more flexibility to choose the windows of integration used in `get_line_intensity`

Add a way to visualise it

Note
------

Merge with #296

Example
-----------

```python

>>> iw = s.estimate_integration_windows()
>>> s.plot(True, integration_windows=iw)
>>> s.get_lines_intensity(integration_windows=iw, plot_result=True)
Cu_Ka at 8.0478 keV : Intensity = 4361.00
Mn_Ka at 5.8987 keV : Intensity = 17007.00

>>> s.plot(True, xray_lines=['Mn_Ka'], integration_windows='auto')

```

eds_integration_windows

### CODE REVIEW

File: hyperspy/_signals/eds.py [hunk 1]
```diff
@@ -549,17 +554,17 @@ def get_lines_intensity(self,
...
         # signal_to_index = self.axes_manager.navigation_dimension - 2
-        for Xray_line in xray_lines:
+        for Xray_line, window in zip(xray_lines, integration_windows):
             line_energy, line_FWHM = self._get_line_energy(Xray_line,
                                                            FWHM_MnKa='auto')
             element, line = utils_eds._get_element_and_line(Xray_line)
-            det = integration_window_factor * line_FWHM / 2.
-            img = self[..., line_energy - det:line_energy + det
-                       ].integrate1D(-1)
+            img = self[..., window[0]:window[1]].integrate1D(-1)
```
Comment (to266) [inline hyperspy/_signals/eds.py]: I guess here you want to slice the signal? as per #427 , maybe then better to use `.isig[]` explicitly? It's your call in the end, though

File: hyperspy/_signals/eds.py [hunk 2]
```diff
@@ -618,3 +623,187 @@ def get_take_off_angle(self):
...
+        Add vertical markers for each X-ray lines.
+
+        Parameters
+        ----------
+        position: 2D array of float
+            The position of the lines in energy. Each row corresponds to a
+            X-ray lines.
+        kwargs
+            keywords argument for markers.vertical_line
+        """
```
Comment (to266) [inline hyperspy/_signals/eds.py]: As per our discussion, maybe rename to `_add_vertical_line_groups`? 

In the documentation, maybe changing "X-ray" to "group (of)" is sufficient?

File: hyperspy/_signals/eds.py [hunk 3]
```diff
@@ -618,3 +623,187 @@ def get_take_off_angle(self):
...
+
+        line_energy = []
+        intensity = []
+        for xray_line in xray_lines:
+            element, line = utils_eds._get_element_and_line(xray_line)
+            line_energy.append(self._get_line_energy(xray_line))
+            relative_factor = elements_db[element][
+                'Atomic_properties']['Xray_lines'][line]['weight']
+            a_eng = self._get_line_energy(element + '_' + line[0] + 'a')
+            intensity.append(self[..., a_eng].data * relative_factor)
```
Comment (to266) [inline hyperspy/_signals/eds.py]: Again, you may (or may not) want to use the `.isig[]` slicing here

File: hyperspy/_signals/eds.py (hunk 2 above)
Comment (pburdet) [inline hyperspy/_signals/eds.py]: Done

File: hyperspy/_signals/eds.py (hunk 3 above)
Comment (pburdet) [inline hyperspy/_signals/eds.py]: Done

File: hyperspy/_signals/eds.py (hunk 1 above)
Comment (pburdet) [inline hyperspy/_signals/eds.py]: Done, it's better with isig

**Pull request merged**